src/simulator.c

```diff
@@ -0,0 +1,218 @@
+        *backend_name = toupper(c);
+        *backend_name++;
+    }
+
+    while ((b = *supported_backends++) != NULL) {
+        if (strcmp(b, name) == 0) {
+            return 1;
+        }
+    }
+
+    return 0;
+}
+
+/**
+ * @brief Print LVGL version
+ */
+static void pr_lvgl_version(void)
+{
+    fprintf(stdout, "%d.%d.%d-%s\n",
+            LVGL_VERSION_MAJOR,
+            LVGL_VERSION_MINOR,
+            LVGL_VERSION_PATCH,
+            LVGL_VERSION_INFO);
+}
+
+/**
+ * @brief Print usage information
+ */
+static void pr_usage(void)
+{
+    fprintf(stdout, "\nlvglsim [-V] [-B] [-b backend_name] [-W window_width] [-H window_height]\n\n");
+    fprintf(stdout, "-V print LVGL version\n");
+    fprintf(stdout, "-B list supported backends\n");
+}
+
+/**
+ * @brief Configure simulator
+ * @description process arguments recieved by the program to select
+ * appropriate options
+ * @param argc the count of arguments in argv
+ * @param argv The arguments
+ */
+static void configure_simulator(int argc, char **argv)
+{
+    int opt = 0;
+    char *backend_name;
+    char **supported_backends;
+
+    supported_backends = lv_driver_backends_get_supported();
+
+    /* Default values */
+    settings.window_width = atoi(getenv("LV_SIM_WINDOW_WIDTH") ? : "800");
+    settings.window_height = atoi(getenv("LV_SIM_WINDOW_HEIGHT") ? : "480");
+
+    /* Parse the command-line options. */
+    while ((opt = getopt (argc, argv, "b:fmW:H:BVh")) != -1) {
+        switch (opt) {
+        case 'h':
+            pr_usage();
+            exit(EXIT_SUCCESS);
+            break;
+        case 'V':
+            pr_lvgl_version();
+            exit(EXIT_SUCCESS);
+            break;
+        case 'B':
+            pr_supported_backends(supported_backends);
+            exit(EXIT_SUCCESS);
+            break;
+        case 'b':
+            if (backend_exists(supported_backends, optarg) == 0) {
+                die("error no such backend: %s\n", optarg);
+            }
+            selected_backend = strdup(optarg);
+            break;
+        case 'W':
+            settings.window_width = atoi(optarg);
+            break;
+        case 'H':
+            settings.window_height = atoi(optarg);
+            break;
+        case ':':
+            pr_usage();
+            die("Option -%c requires an argument.\n", optopt);
+            break;
+        case '?':
+            pr_usage();
+            die("Unknown option -%c.\n", optopt);
+        }
+    }
+
+    if (selected_backend == NULL) {
+        /* Select the default backend */
+        selected_backend = *supported_backends;
+    }
+}
+
+/**
+ * @brief entry point
+ * @description start a demo
+ * @param argc the count of arguments in argv
+ * @param argv The arguments
+ */
+int main(int argc, char **argv)
+{
+
+    configure_simulator(argc, argv);
+
+    /* Initialize LVGL. */
+    lv_init();
+
+    /* Initialize the configured backend */
+    if (lv_driver_backends_init_backend(selected_backend) == -1) {
```

Comment:
```suggestion
    if (backend_init(selected_backend) == -1) {
```

It's not an LVGL function so it shouldn't start with `lv_*`

Reply:
Okay I will remove the ``lv_`` prefixes for now, but the ultimate goal would be to add a new file ``win32.c`` in ``src/lib/display_backends`` to provide support for Windows GDI. And then it can be turned into a simulator library that is easy to use on all majors OS. At this point we could include it in the main library.